packages/devtools/lib/src/utils.dart

```diff
@@ -145,6 +145,40 @@ String longestFittingSubstring(
   return originalText.substring(0, i);
 }
 
+/// Returns a trimmed vm service uri without any trailing characters.
+///
+/// For example, given a [value] of http://127.0.0.1:60667/72K34Xmq0X0=/#/vm,
+/// this method will return the URI http://127.0.0.1:60667/72K34Xmq0X0=/.
+Uri getTrimmedUri(String value) {
+  final startingUri = Uri.parse(value);
+
+  // Path should be of the form "/72K34Xmq0X0=/", but it could have trailing
+  // characters that we should trim.
+  final path = startingUri.path;
+
+  // Find the end of token index (index of the second slash).
+  int endOfTokenIndex = -1;
```

Comment:
you could split on '/' 
and then build the path back up by
parts.take(3).join('/')

Reply:
done.